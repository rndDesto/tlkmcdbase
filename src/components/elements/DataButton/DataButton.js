import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';

const DataButton = (props) => {
  const {
    buttonProps,
    children,
    onClick,
    id,
    isLoading } = props;
  return (
    <Button
      id={id}
      {...buttonProps}
      onClick={onClick}>
      {isLoading ? 'Mengirim':children}</Button>
  );
};

DataButton.defaultProps = {
  buttonProps: {},
  children: null,
  id:'',
  isLoading:false,
  onClick: () => {},
};

DataButton.propTypes = {
  buttonProps: PropTypes.object,
  children: PropTypes.node,
  id:PropTypes.string,
  isLoading: PropTypes.bool,
  onClick: PropTypes.func,
};

export default DataButton;
