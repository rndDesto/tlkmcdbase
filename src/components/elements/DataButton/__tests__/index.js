import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import DataButton from '../DataButton';

describe('src/components/elements/DataButton', () => {

  test('render', () => {
    const shallow = new ShallowRenderer();
    DataButton.defaultProps.onClick();
    const tree = shallow.render(<DataButton />);
    expect(tree).toMatchSnapshot();
  });

  test('isLoading', () => {
    const shallow = new ShallowRenderer();
    DataButton.defaultProps.onClick();
    const tree = shallow.render(<DataButton isLoading={true}/>);
    expect(tree).toMatchSnapshot();
  });

});

