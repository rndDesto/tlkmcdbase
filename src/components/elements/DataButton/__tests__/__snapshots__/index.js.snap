// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/DataButton isLoading 1`] = `
<ForwardRef(WithStyles)
  id=""
  onClick={[Function]}
>
  Mengirim
</ForwardRef(WithStyles)>
`;

exports[`src/components/elements/DataButton render 1`] = `
<ForwardRef(WithStyles)
  id=""
  onClick={[Function]}
/>
`;
